src/runtime/lpython_builtin.py

```diff
@@ -314,3 +314,26 @@ def _lpython_imag(x: c64) -> f64:
 @overload
 def _lpython_imag(x: c32) -> f32:
     return _lfortran_caimag(x)
+
+
+# FloorDiv operator (//)
+
+@overload
+def _lpython_floordiv(a: i32, b: i32) -> i32:
+    return int(a/b)
+
+@overload
+def _lpython_floordiv(a: i32, b: i64) -> i64:
+    return int(a/b)
+
+@overload
+def _lpython_floordiv(a: i64, b: i32) -> i64:
+    return int(a/b)
+
+@overload
+def _lpython_floordiv(a: i64, b: i64) -> i64:
+    return int(a/b)
+
+@overload
+def _lpython_floordiv(a: f64, b: f64) -> f64:
+    return floor(int(a/b))
```

Comment:
I think it has to be just:
```suggestion
    return floor(a/b)
```
No cast to `int` first.

Reply:
I would just keep the following 4 functions:
* `def _lpython_floordiv(a: i32, b: i32) -> i32`
* `def _lpython_floordiv(a: i64, b: i64) -> i64`
* `def _lpython_floordiv(a: f32, b: f32) -> f32`
* `def _lpython_floordiv(a: f64, b: f64) -> f64`

And the ast2asr frotnend would do the correct casting first of int to float, or of 32 to 64 bit.

Reply:
> I think it has to be just:

oh, it's a typo, I meant `float(int(a/b))`. Thanks. using `floor` hangs in ASR (maybe multiple module issue)